# Project Title: Diabetes Prediction Analysis
## Section 3: Hypothesis Testing and Statistical Analysis
## Course: CSC 405 - Data Science
## Team Members: Adrian Aldridge, Eman Dweik, Sarah Robinson
## Dataset Source:
https://www.kaggle.com/datasets/miadul/synthetic-diabetes-prediction-dataset

## First Test: Two-sided T-Test on Diabetes Status and Glucose Levels
### Question
Do individuals with diabetes have significantly higher glucose levels compared to those without diabetes?
### Hypotheses
**Null Hypothesis(H₀)**: There is no difference in mean glucose levels between diabetic and non-diabetic individuals (μ_diabetic = μ_non-diabetic)

**Alternate Hypothesis(H₁)**: There is a difference in mean glucose levels between the two groups (μ_diabetic ≠ μ_non-diabetic)
### Justification
We are using a t-test to determine whether mean glucose levels differ between two independent groups: individuals with diabetes and those without. An independent sample t-test is appropriate here because the two groups (diabetic and non-diabetic) are independent, no pairing or repeated tests. The t-test is the most appropriate for comparing the means of two groups. 
### Code and Testing

In [7]:
#Import modules
import pandas as pd
from scipy.stats import ttest_ind

#Read in dataset as dataframe
df = pd.read_csv("diabetes_clean.csv")

#Separate glucose values into two groups based on diabetes status
#Group 1 - Individuals with diabetes
diabetic = df[df['diabetes'] == 1]['glucose']
#Group 2 - Individuals without diabetes
non_diabetic = df[df['diabetes'] == 0]['glucose']

#Running t-test (does not assume equal variance)
t_stat, p_value = ttest_ind(diabetic, non_diabetic, equal_var=False)

#Printing statistics
print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.6f}")

T-statistic: -0.6227
P-value: 0.533522


### Interpretation
**Level of significance**: 0.05

**Reported P-value**: 0.53

The P-value produced is much larger than our level of significance, leading us to fail to reject the null hypothesis. Meaning, there is no statistically significant difference in glucose levels between diabetic and non-diabetic individuals in this dataset. 

This result could be due to the use of a synthetic dataset, the glucose values could have a high variance leading to overlap in glucose readings between the two groups. This does not reflect real world data where glucose levels differ greatly between diabetic and non-diabetic individuals.

## Second Test: Chi-Square on Gender vs. Diabetes
### Question: 
Is there an association between gender and diabetes?
### Hypotheses
**Null Hypothesis(H₀)**: Gender and diabetes are independent 

**Alternate Hypothesis(H₁)**: Gender and diabetes are associated
### Justification
This test examines whether there is a statistical association between two categorical variables: gender and diabetes status. The chi square test is appropriate because the data can be summarized in a contingency table of observed frequencies. The goal is to test whether the distribution of diabetes status differs across gender categories. The chi square test is the most appropriate method for determining whether two categorical variables are statistically independent.
### Code and Testing

In [8]:
#Import modules
from scipy.stats import chi2_contingency

#Create contingency table counting the occurences of each combination
contingency_table = pd.crosstab(df['gender'], df['diabetes'])

#Run chi-square test
chi2, p, dof, expected = chi2_contingency(contingency_table)

#Print statistics
print(f"Chi-square statistic: {chi2:.4f}")
print(f"P-value: {p:.6f}")

Chi-square statistic: 2.2888
P-value: 0.130308


### Interpretation
**Level of significance**: 0.05

**Reported P-value**: 0.13

The P-value produced is larger than our significance level, therefore we fail to reject the null hypothesis. Meaning, there is no statistically significant association between gender and diabetes. Diabetes status appears to be independent of gender in this particular dataset. 

These results could imply that diabetes risk is entirely gender independent, meaning other factors are more likely to predict diabetes status. We must remember this dataset is synthetic, and while it does not consist of real world data points this particular result does reflect real world data in that diabetes is entirely independent of ones gender.

## Third Test: Pearson Correlation on BMI vs. Glucose
### Question: 
Is there a relationship between BMI and glucose levels?
### Hypotheses
**Null Hypothesis(H₀)**: No linear correlation between BMI and glucose.

**Alternate Hypothesis(H₁)**: A linear correlation exists between BMI and glucose.
### Justification
This test is to investigate the relationship between BMI and glucose. The Pearson correlation coefficient is chosen because, the goal is to measure the strength and direction of a linear relationship between two continuous variables. Pearson correlation quantifies this on a scale from -1 to +1. Pearson correlation is the most appropriate test to evaluate the linear relationship between two continuous variables.
### Code and Testing

In [6]:
#Import modules
from scipy.stats import pearsonr

#Perform pearson correlation
corr, p_value = pearsonr(df['bmi'], df['glucose'])

#Print statistics
print(f"Correlation coefficient: {corr:.4f}")
print(f"P-value: {p_value:.6f}")

Correlation coefficient: 0.0025
P-value: 0.799629


### Interpretation
For a Pearson correlation test the correlation ranges are:
* ~0 → no relationship
* ±0.3 → weak
* ±0.5 → moderate
* ±0.7+ → strong

**Correlation Coefficient**: 0.0025
The correlation coefficient given is extremely close to 0, implying a very weak to no linear correlation between BMI and glucose levels in this dataset. Leading us to infer that BMI does not affect glucose levels.

This could be due to the fact that this is a synthetic dataset and not reflective of real world data, which implies quite the opposite that BMI and glucose levels are heavily correlated. 

### Conclusions
Individual variable tests show weak or no statistical relationships. This aligns with the exploratory analysis concluded in Section 2 - Data Cleaning and Exploratory Analysis. Almost no correlations were found between data points. This is most likely caused due to the synthetic nature of the dataset and the inherent complexity and multi-variability of diabetes predicition. A lack of strong univariate relationships implies that a multivariate technique such as logistic regression or machine learning may be needed to get any kind of results out of this dataset.